modules/Registration/components/student/StudentRegistrationForm.helpers.ts

```diff
@@ -0,0 +1,64 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { z } from "zod";
+
+const educationLevelEnum = z.enum(["School", "College", "University"]);
+
+export const studentRegistrationSchema = z
+  .object({
+    firstName: z.string().min(1, "First name is required"),
+    lastName: z.string().min(1, "Last name is required"),
+    gender: z.string().min(1, "Gender is required"),
+    email: z.string().email("Invalid email"),
+    phoneNumber: z.string().min(1, "Phone number is required"),
+    address: z.string().min(1, "Address is required"),
+    educationLevel: educationLevelEnum.optional().refine((val) => val !== undefined, {
+      message: "Education level is required",
+    }),
+    password: z.string().min(6, "Password must be at least 6 characters"),
+    confirmPassword: z.string(),
+    englishBanglaMedium: z.string().optional(),
+    class: z.string().optional(),
+    degreeType: z.enum(["Bachelors", "Masters"]).optional(),
+    degreeName: z.string().optional(),
+    semesterYear: z.string().optional(),
```

Comment:
Please look into the usage of superRefine and how to make these fields required based on educationLevel field.

Can pair with Faysal bhai as he has implemented this.

Reply:
[
<img width="711" alt="education_level_error" src="https://github.com/user-attachments/assets/5c564892-7ed8-495c-8ca6-71193017676f">
](url)